### PR TITLE
GH-37601: [C++][Parquet] Add missing GoogleMock dependency

### LINKED ISSUE #37601: [CI][C++] test-ubuntu-20.04-cpp-minimal-with-formats failed
### Describe the bug, including details regarding any error messages, version, and platform.

https://github.com/ursacomputing/crossbow/actions/runs/6099155746/job/16551111386#step:6:1592

```text
 [ 92%] Building CXX object src/parquet/CMakeFiles/parquet-file-deserialize-test.dir/file_deserialize_test.cc.o
/arrow/cpp/src/parquet/file_deserialize_test.cc:18:10: fatal error: gmock/gmock.h: No such file or directory
   18 | #include <gmock/gmock.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
```

### Component(s)

C++, Continuous Integration

### PR DESCRIPTION
**Pull request by kou**
### Rationale for this change

Parquet's tests uses GoogleMock. So we should specify GoogleMock as a dependency explicitly.

### What changes are included in this PR?

Add `${ARROW_GTEST_GMOCK}` to `PARQUET_MIN_TEST_LIBS`.

### Are these changes tested?

Yes.

### Are there any user-facing changes?

No.
* Closes: #37601

### CODE REVIEW

**Comment by kou** [comment]
@github-actions crossbow submit test-ubuntu-20.04-cpp-minimal-with-formats

File: cpp/src/parquet/CMakeLists.txt [hunk 1]
```diff
@@ -114,12 +114,13 @@ else()
   set(ARROW_LIBRARIES_FOR_STATIC_TESTS arrow_testing_shared arrow_shared)
 endif()
 
-set(PARQUET_MIN_TEST_LIBS ${ARROW_GTEST_GTEST_MAIN} ${ARROW_GTEST_GTEST} Boost::headers)
+set(PARQUET_MIN_TEST_LIBS ${ARROW_GTEST_GMOCK} ${ARROW_GTEST_GTEST}
```
Comment (mapleFU) [inline cpp/src/parquet/CMakeLists.txt:117]: This patch add a `ARROW_GTEST_GMOCK` here? The latter `list(Append...` is just a style change?

Comment (kou) [reply cpp/src/parquet/CMakeLists.txt:117]: Yes. `ARROW_GTEST_GMOCK` is the main change. `list(APPEND` is for modenizing our CMake code.
It's not related to this fix. I'm modernizing our CMake code step-by-step when I touch related CMake code.
If you don't like it, I can remove the change from this PR.

Comment (mapleFU) [reply cpp/src/parquet/CMakeLists.txt:117]: It's ok, just want to make it clear :-)

**Review by mapleFU — APPROVED** [review]
LGTM

**Review by wgtmac — APPROVED** [review]

**Pull request merged**